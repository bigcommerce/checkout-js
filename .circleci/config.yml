aliases:
  - &node_executor
      executor:
        name: node/node
        node-version: "14.18"

  - &release_filter
      branches:
        only:
          - master

  - &pull_request_filter
      branches:
        ignore:
          - master

version: 2.1

orbs:
  ci: bigcommerce/internal@volatile
  node: bigcommerce/internal-node@volatile

jobs:
  test:
    <<: *node_executor
    steps:
      - ci/pre-setup
      - node/npm-install
      - run:
          name: "Run unit tests"
          command: npm run test -- --coverage --runInBand
      - store_artifacts:
          path: coverage
          destination: coverage

  build:
    <<: *node_executor
    steps:
      - ci/pre-setup
      - node/npm-install
      - run:
          name: "Run linter"
          command: npm run lint
      - run:
          name: "Test build"
          command: npm run build
      - persist_to_workspace:
          root: .
          paths:
            - dist

  e2e:
    <<: *node_executor
    steps:
      - ci/pre-setup
      - node/npm-install
      - run:
          name: "Build checkout-js"
          command: npm run build
      - run:
          name: "Install playwright"
          command: npx playwright install --with-deps
      - run:
          name: "Run e2e tests"
          command: npx playwright test
      - store_artifacts:
          path: packages/payment-integration-test-framework/report
          destination: artifact-file
      - store_artifacts:
          path: packages/payment-integration-test-framework/screenshots

  release:
    <<: *node_executor
    steps:
      - ci/pre-setup
      - node/npm-install
      - run:
          name: "Configure Git user"
          command: |
            git config user.email $GIT_USER_EMAIL
            git config user.name $GIT_USER_NAME
      - run:
          name: "Remove old release files"
          command: rm -rf dist
      - attach_workspace:
          at: .
      - run:
          name: "Create and tag new release"
          command: npm run release:version
      - run:
          name: "Push commits and tag to GitHub"
          command: |
            git push --follow-tags origin $CIRCLE_BRANCH
      - run:
          name: "Export release version to file"
          command: |
            echo "export RELEASE_VERSION=$(git describe --abbrev=0)" > /tmp/release_version.txt
            echo "export RELEASE_REVISION=$(git rev-parse HEAD)" >> /tmp/release_version.txt
      - persist_to_workspace:
          root: /tmp
          paths:
            - release_version.txt

  upload_to_sentry:
    <<: *node_executor
    steps:
      - ci/pre-setup
      - run:
          name: "Install Sentry cli"
          command: curl -sL https://sentry.io/get-cli/ | bash
      - attach_workspace:
          at: /tmp
      - run:
          name: "Upload source maps to Sentry"
          command: |
            source /tmp/release_version.txt
            git checkout $RELEASE_VERSION
            SENTRY_RELEASE=$SENTRY_PROJECT@${RELEASE_VERSION#v}
            SENTRY_COMMIT=$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME@$RELEASE_REVISION
            sentry-cli releases --project $SENTRY_PROJECT --org $SENTRY_ORG new $SENTRY_RELEASE
            sentry-cli releases --project $SENTRY_PROJECT --org $SENTRY_ORG set-commits $SENTRY_RELEASE --commit $SENTRY_COMMIT
            sentry-cli releases --project $SENTRY_PROJECT --org $SENTRY_ORG files $SENTRY_RELEASE upload-sourcemaps dist
            sentry-cli releases --project $SENTRY_PROJECT --org $SENTRY_ORG finalize $SENTRY_RELEASE

workflows:
  version: 2

  pull_request:
    jobs:
      - test:
          filters:
            <<: *pull_request_filter
      - build:
          filters:
            <<: *pull_request_filter
      - e2e:
          filters:
            <<: *pull_request_filter
      - ci/validate-commits:
          filters:
            <<: *pull_request_filter

  release:
    jobs:
      - test:
          filters:
            <<: *release_filter
      - build:
          filters:
            <<: *release_filter
      - e2e:
          filters:
            <<: *release_filter
      - approve:
          filters:
            <<: *release_filter
          type: approval
          requires:
            - test
            - build
            - e2e
      - release:
          filters:
            <<: *release_filter
          requires:
            - approve
      - upload_to_sentry:
          filters:
            <<: *release_filter
          requires:
            - approve
            - release
