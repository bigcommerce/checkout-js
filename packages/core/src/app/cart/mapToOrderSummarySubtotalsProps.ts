import { Checkout } from '@bigcommerce/checkout-sdk';

import { OrderSummarySubtotalsProps } from '../order';
import { hasSelectedShippingOptions } from '../shipping';

export default function mapToOrderSummarySubtotalsProps({
    subtotal,
    cart: { discountAmount },
    giftCertificates,
    consignments,
    handlingCostTotal,
    shippingCostBeforeDiscount,
    giftWrappingCostTotal,
    coupons,
    taxes,
}: Checkout): OrderSummarySubtotalsProps {
    return {
        subtotalAmount: subtotal,
        discountAmount,
        giftCertificates,
        giftWrappingAmount: giftWrappingCostTotal,
        shippingAmount: hasSelectedShippingOptions(consignments)
            ? shippingCostBeforeDiscount
            : undefined,
        handlingAmount: handlingCostTotal,
        coupons,
        taxes,
    };
}
