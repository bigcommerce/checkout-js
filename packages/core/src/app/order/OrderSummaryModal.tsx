import {
    LineItemMap,
    ShopperCurrency as ShopperCurrencyType,
    StoreCurrency,
} from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent, ReactNode } from 'react';

import { preventDefault } from '../common/dom';
import { TranslatedString } from '../locale';
import { IconClose } from '../ui/icon';
import { Modal, ModalHeader } from '../ui/modal';

import OrderSummaryItems from './OrderSummaryItems';
import OrderSummarySection from './OrderSummarySection';
import OrderSummarySubtotals, { OrderSummarySubtotalsProps } from './OrderSummarySubtotals';
import OrderSummaryTotal from './OrderSummaryTotal';

export interface OrderSummaryDrawerProps {
    additionalLineItems?: ReactNode;
    lineItems: LineItemMap;
    total: number;
    storeCurrency: StoreCurrency;
    shopperCurrency: ShopperCurrencyType;
    isOpen: boolean;
    headerLink?: ReactNode;
    onRequestClose?(): void;
    onAfterOpen?(): void;
}

const OrderSummaryModal: FunctionComponent<
    OrderSummaryDrawerProps & OrderSummarySubtotalsProps
> = ({
    additionalLineItems,
    children,
    onRequestClose,
    onAfterOpen,
    storeCurrency,
    shopperCurrency,
    isOpen,
    headerLink,
    lineItems,
    total,
    ...orderSummarySubtotalsProps
}) => (
    <Modal
        additionalBodyClassName="cart-modal-body optimizedCheckout-orderSummary"
        additionalHeaderClassName="cart-modal-header optimizedCheckout-orderSummary"
        header={renderHeader({ headerLink, onRequestClose })}
        isOpen={isOpen}
        onAfterOpen={onAfterOpen}
        onRequestClose={onRequestClose}
    >
        <OrderSummarySection>
            <OrderSummaryItems items={lineItems} />
        </OrderSummarySection>
        <OrderSummarySection>
            <OrderSummarySubtotals {...orderSummarySubtotalsProps} />
            {additionalLineItems}
        </OrderSummarySection>
        <OrderSummarySection>
            <OrderSummaryTotal
                orderAmount={total}
                shopperCurrencyCode={shopperCurrency.code}
                storeCurrencyCode={storeCurrency.code}
            />
        </OrderSummarySection>
    </Modal>
);

const renderHeader: FunctionComponent<{
    headerLink: ReactNode;
    onRequestClose?(): void;
}> = ({ onRequestClose, headerLink }) => (
    <>
        <a className="cart-modal-close" href="#" onClick={preventDefault(onRequestClose)}>
            <span className="is-srOnly">
                <TranslatedString id="common.close_action" />
            </span>
            <IconClose />
        </a>
        <ModalHeader additionalClassName="cart-modal-title">
            <TranslatedString id="cart.cart_heading" />
        </ModalHeader>

        {headerLink}
    </>
);

export default OrderSummaryModal;
