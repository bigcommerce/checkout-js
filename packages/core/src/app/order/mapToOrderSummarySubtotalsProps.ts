import { Order } from '@bigcommerce/checkout-sdk';

import { mapFromPayments } from '../giftCertificate';

import getStoreCreditAmount from './getStoreCreditAmount';
import { OrderSummarySubtotalsProps } from './OrderSummarySubtotals';

export default function mapToOrderSummarySubtotalsProps({
    baseAmount,
    discountAmount,
    shippingCostBeforeDiscount,
    payments,
    handlingCostTotal,
    giftWrappingCostTotal,
    coupons,
    taxes,
}: Order): OrderSummarySubtotalsProps {
    return {
        subtotalAmount: baseAmount,
        shippingAmount: shippingCostBeforeDiscount,
        giftWrappingAmount: giftWrappingCostTotal,
        discountAmount,
        storeCreditAmount: getStoreCreditAmount(payments),
        handlingAmount: handlingCostTotal,
        coupons,
        giftCertificates: payments && mapFromPayments(payments),
        taxes,
    };
}
