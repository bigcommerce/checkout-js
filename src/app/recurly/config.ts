export const recurlyId = window.RECURLY_PUBLIC_KEY ?? process.env.RECURLY_PUBLIC_KEY;

// @ts-ignore
export const apiEndpoint = window.FRONTEND_API_ENDPOINT ?? process.env.FRONTEND_API_ENDPOINT;
// @ts-ignore
export const recurlyBcAppId = window.RECURLY_BC_APP_ID ?? process.env.RECURLY_BC_APP_ID;
export default {recurlyId, apiEndpoint, recurlyBcAppId};
