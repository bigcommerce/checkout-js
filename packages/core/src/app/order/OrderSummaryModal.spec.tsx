import { Order } from '@bigcommerce/checkout-sdk';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { getStoreConfig } from '../config/config.mock';

import mapToOrderSummarySubtotalsProps from './mapToOrderSummarySubtotalsProps';
import { getOrder } from './orders.mock';
import OrderSummaryModal from './OrderSummaryModal';

let order: Order;
let orderSummary: ShallowWrapper;

describe('OrderSummaryModal', () => {
    beforeEach(() => {
        order = getOrder();

        orderSummary = shallow(
            <OrderSummaryModal
                isOpen={true}
                {...mapToOrderSummarySubtotalsProps(order)}
                additionalLineItems="foo"
                lineItems={order.lineItems}
                shopperCurrency={getStoreConfig().shopperCurrency}
                storeCurrency={getStoreConfig().currency}
                total={order.orderAmount}
            />,
        );
    });

    it('renders order summary', () => {
        expect(orderSummary).toMatchSnapshot();
    });
});
