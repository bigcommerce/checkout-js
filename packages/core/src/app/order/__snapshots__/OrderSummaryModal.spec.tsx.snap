// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummaryModal renders order summary 1`] = `
<Modal
  additionalBodyClassName="cart-modal-body optimizedCheckout-orderSummary"
  additionalHeaderClassName="cart-modal-header optimizedCheckout-orderSummary"
  header={
    <React.Fragment>
      <a
        className="cart-modal-close"
        href="#"
        onClick={[Function]}
      >
        <span
          className="is-srOnly"
        >
          <WithLanguage(TranslatedString)
            id="common.close_action"
          />
        </span>
        <Memo />
      </a>
      <ModalHeader
        additionalClassName="cart-modal-title"
      >
        <WithLanguage(TranslatedString)
          id="cart.cart_heading"
        />
      </ModalHeader>
    </React.Fragment>
  }
  isOpen={true}
>
  <OrderSummarySection>
    <OrderSummaryItems
      items={
        Object {
          "customItems": Array [],
          "digitalItems": Array [],
          "giftCertificates": Array [
            Object {
              "amount": 100,
              "id": "bd391ead-8c58-4105-b00e-d75d233b429a",
              "message": "message",
              "name": "$100 Gift Certificate",
              "recipient": Object {
                "email": "lu@is.com",
                "name": "luis",
              },
              "sender": Object {
                "email": "pa@blo.com",
                "name": "pablo",
              },
              "taxable": false,
              "theme": "General",
            },
          ],
          "physicalItems": Array [
            Object {
              "addedByPromotion": false,
              "brand": "OFS",
              "categoryNames": Array [
                "Cat 1",
              ],
              "comparisonPrice": 200,
              "couponAmount": 0,
              "discountAmount": 0,
              "discounts": Array [],
              "extendedComparisonPrice": 250,
              "extendedListPrice": 200,
              "extendedSalePrice": 200,
              "id": "666",
              "imageUrl": "/images/canvas-laundry-cart.jpg",
              "isShippingRequired": true,
              "isTaxable": true,
              "listPrice": 200,
              "name": "Canvas Laundry Cart",
              "options": Array [
                Object {
                  "name": "n",
                  "nameId": 1,
                  "value": "v",
                  "valueId": 3,
                },
              ],
              "productId": 103,
              "quantity": 1,
              "salePrice": 200,
              "sku": "CLC",
              "url": "/canvas-laundry-cart/",
              "variantId": 71,
            },
          ],
        }
      }
    />
  </OrderSummarySection>
  <OrderSummarySection>
    <Memo(OrderSummarySubtotals)
      coupons={
        Array [
          Object {
            "code": "savebig2015",
            "couponType": "percentage_discount",
            "discountedAmount": 5,
            "displayName": "20% off each item",
            "id": "1",
          },
          Object {
            "code": "279F507D817E3E7",
            "couponType": "shipping_discount",
            "discountedAmount": 5,
            "displayName": "$5.00 off the shipping total",
            "id": "4",
          },
        ]
      }
      discountAmount={10}
      giftCertificates={
        Array [
          Object {
            "balance": 10,
            "code": "gc",
            "purchaseDate": "",
            "remaining": 3,
            "used": 7,
          },
        ]
      }
      giftWrappingAmount={0}
      handlingAmount={8}
      shippingAmount={20}
      storeCreditAmount={0}
      subtotalAmount={200}
      taxes={
        Array [
          Object {
            "amount": 3,
            "name": "Tax",
          },
        ]
      }
    />
    foo
  </OrderSummarySection>
  <OrderSummarySection>
    <WithCurrency(OrderSummaryTotal)
      orderAmount={190}
      shopperCurrencyCode="USD"
      storeCurrencyCode="USD"
    />
  </OrderSummarySection>
</Modal>
`;
